src/main/resources/com/tresys/pcap/xsd/pcap.dfdl.xsd

```diff
@@ -370,9 +370,12 @@ The message root is 'PCAP'.
           </xs:complexType>
         </xs:element>
         <xs:element name="Protocol" type="xs:unsignedInt" dfdl:inputValueCalc="{ xs:unsignedInt(../IPv4Header/Protocol) }" />
-        <xs:choice>
-            <xs:element ref="pcap:TransportLayer" dfdl:lengthKind="explicit" dfdl:lengthUnits="bytes" dfdl:length="{ xs:unsignedInt(../IPv4Header/Length) - 20 }" />
-            <xs:element ref="pcap:ICMPv4" dfdl:lengthKind="explicit" dfdl:lengthUnits="bytes" dfdl:length="{ xs:unsignedInt(../IPv4Header/Length) - 20 }"/>
+        <xs:choice dfdl:choiceDispatchKey="{
+          if (./Protocol eq 1) then 'ICMPv4'
+          else 'other'
+           }">
```

Comment:
Same comment as above. And another nice use case for choiceBranchDefault.

Reply:
In this case, we have 1 for icmpv4, and 6 and 17 for tcp and udp respectively. So we don't need "other".